src/pages/patientView/clinicalInformation/CohortVariantCountCache.spec.ts

```diff
@@ -0,0 +1,273 @@
+import {default as CohortVariantCountCache, EntrezToKeywordList, VariantCountCacheType} from './CohortVariantCountCache';
+import { assert, default as chai } from 'chai';
+import sinon from 'sinon';
+import {VariantCount} from "../../../shared/api/CBioPortalAPIInternal";
+
+chai.config.truncateThreshold = 0;
+
+
+describe('CohortVariantCountCache', () => {
+
+    let data:VariantCount[];
+
+    let sampleIds:string[];
+
+    let fetchPromise:Promise<VariantCount[]>;
+
+    let initialCache:VariantCountCacheType = {
+        numberOfSamples: null,
+        mutationInGene: {},
+        keyword: {}
+    };
+
+    class TestCache extends CohortVariantCountCache {
```

Comment:
I think that this can be accomplished better with Sinon.  If I follow correctly, what we're testing is that fetch is called with the profileId passed to constructor.  That could be accomplished by stubbing the fetch method of the class itself.  We can then use sinon properties (calledOnce etc) and args collection.   I think it's clearer and more flexibly if we monkey with the prototype of the class rather than extending it in this way.  Lets discuss.